Public/Src/Tools/UnitTests/Analyzers/XLGToDBAnalyzerTests.cs

```diff
@@ -69,5 +74,62 @@ public void DBCreatedAndPopulated()
 
             XAssert.AreNotEqual(Directory.GetFiles(OutputDirPath.ToString(Context.PathTable), "*.sst").Length, 0);
         }
+
+        /// <summary>
+        /// This test makes sure that the number of events logged (per event type) match 
+        /// what is expected based on the build schedule that we manually populate and build.
+        /// </summary>
+        [Fact]
+        public void DBEventCountVerification()
+        {
+            Configuration.Logging.LogExecution = true;
+
+            var fileOne = FileArtifact.CreateOutputFile(Combine(TestDirPath, "foo.txt"));
+            var fileTwo = FileArtifact.CreateOutputFile(Combine(TestDirPath, "bar.txt"));
+            var dirOne = CreateOutputDirectoryArtifact(Combine(TestDirPath, "baz").ToString(Context.PathTable));
+            Directory.CreateDirectory(ArtifactToString(dirOne));
+
+            var pipA = CreateAndSchedulePipBuilder(new Operation[]
+            {
+                Operation.WriteFile(fileOne),
+            }).Process;
+
+            var pipB = CreateAndSchedulePipBuilder(new Operation[]
+            {
+                Operation.WriteFile(fileTwo),
+            }).Process;
+
+            var pipC = CreateAndSchedulePipBuilder(new Operation[]
+            {
+                Operation.ReadFile(fileOne),
+                Operation.WriteFile(CreateOutputFileArtifact())
+            }).Process;
+
+            var pipD = CreateAndSchedulePipBuilder(new Operation[]
+            {
+                Operation.EnumerateDir(dirOne),
+                Operation.WriteFile(CreateOutputFileArtifact())
+            }).Process;
+
+            var buildA = RunScheduler().AssertSuccess();
+
+            var analyzerRes = RunAnalyzer(buildA).AssertSuccess();
+
+            var dataStore = new XldbDataStore(storeDirectory: OutputDirPath.ToString(Context.PathTable));
+
+            XAssert.AreEqual(dataStore.GetFileArtifactContentDecidedEvents().Count(), 5);

```

Comment:
Why is 5 expected here?  I see only 4 output files in this test.

More generally speaking, instead of using a constant number here, I'd like to use some expression instead.  For example, after calling `RunScheduler()`, pip graph is created and saved in `SchedulerIntegrationTestBase.m_lastGraph` field.  That field is currently private, but you could add a protected getter property for it, and then here instead of `5` say something like `LastGraph.AllFiles.Count()` (or whatever the logic should be)